### PR TITLE
Clean Up

### PR DESCRIPTION
**Pull request by blink1073**
Clean up style, reorganize tests.
Fix scoping errors.

### CODE REVIEW

File: test/src/mocksocket.ts [hunk 1]
```diff
@@ -82,21 +77,21 @@ class SocketBase {
...
-    if (this._onClose) {
-      utils.doLater(() => {
-        this._readyState = SocketBase.CLOSED;
-        this._onClose();
-      });
-    }
+    utils.doLater(() => {
+      this._readyState = SocketBase.CLOSED;
+      var onClose = this._onClose;
+      if (onClose) onClose();
```
Comment (rgbkrk) [inline test/src/mocksocket.ts]: This one has different logic than before, changing the `_readyState` whether or not `this._onClose` is defined.

Comment (rgbkrk) [inline test/src/mocksocket.ts]: (Just a note - I have no idea what the right behavior is here)

**Comment by rgbkrk** [comment]
These are all for the mocks, so it's all fine to me.

I'm also ok with picking a style and sticking to it with either an eslintrc, jshintrc, or even one of the `*standard*` variants. I'd open that up on the mailing list, but it's not my can of worms.

File: test/src/mocksocket.ts (hunk 1 above)
Comment (blink1073) [inline test/src/mocksocket.ts]: It is correct in this PR.

**Pull request merged**